src/builder/create_message.rs

```diff
@@ -55,57 +55,66 @@ impl<'a> CreateMessage<'a> {
     ///
     /// **Note**: Message contents must be under 2000 unicode code points.
     #[inline]
-    pub fn content<D: Display>(&mut self, content: D) {
+    pub fn content<D: Display>(&mut self, content: D) -> &mut Self {
         self._content(content.to_string());
+        self
     }
 
     fn _content(&mut self, content: String) {
         self.0.insert("content", Value::String(content));
     }
 
     /// Set an embed for the message.
-    pub fn embed<F: FnOnce(CreateEmbed) -> CreateEmbed>(&mut self, f: F) {
-        let map = utils::vecmap_to_json_map(f(CreateEmbed::default()).0);
+    pub fn embed<F: FnOnce(&mut CreateEmbed) -> &mut CreateEmbed>(&mut self, f: F) -> &mut Self {
```

Comment:
Use the `where` syntax:
```suggestion
    pub fn embed<F>(&mut self, f: F) -> &mut Self 
        where F: FnOnce(&mut CreateEmbed) -> &mut CreateEmbed {
```